feat(Gate): add business partner relations

nicoprow opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description
<!-- 
Please describe your PR: 
- What does this PR introduce? 
- Does it fix a bug? 
- Does it add a new feature?
- Is it enhancing documentation?
-->

This pull request adds  new endpoints to search, create, update and delete business partner relations.
It currently supports only the IsManagedBy relation type.

- add logic for CRUD and constraint checking
- add database table for relations
- add tests

<!-- Please tag the related issue `Fixes or Updates #issue_number`, if applicable. -->

Contributes to #1027

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

SujitMBRDI reviewed (changes requested):
Great logic and code changes!!
Changes looks good to me, tests i have performed: 
1. From gate input, I have provided two legal entities 
2. Then created relations using tenant null
3. Then deleted relationship too.

As we discussed in review session, just we need to swap logic for `Source` and `Target` as `IsManagedBy` is passive term.

nicoprow commented:
I changed the constraints on the IsManagedBy relation as discussed.
Also, I removed the tenantBpnL as it caused confusion before. Maybe at some later point we can add it again as a hidden parameter that only appears on no-auth profiles. But that is quite extra work that I feel warrants its own contribution.
